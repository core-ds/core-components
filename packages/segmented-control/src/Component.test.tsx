import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';

import { Segment, SegmentedControl, SegmentedControlProps } from './index';

const SEGMENT_TEST_ID = 'SEGMENT';

const renderComponent = ({
    selectedId,
    onChange,
    ...restProps
}: Omit<SegmentedControlProps, 'children'>) => (
    <SegmentedControl onChange={onChange} selectedId={selectedId} {...restProps}>
        <Segment id={1} title={'Label 1'} />
        <Segment dataTestId={SEGMENT_TEST_ID} id={2} title={'Label 2'} />
        <Segment id={3} title={'Label 3'} />
        <Segment id={4} title={'Label 4'} />
    </SegmentedControl>
);

describe('segmented-control', () => {
    it('should display correctly', () => {
        const { container } = render(renderComponent({ onChange: () => null, selectedId: 1 }));
        expect(container).toMatchSnapshot();
    });

    it('should onChange get correct id', () => {
        const onChange = jest.fn();
        render(
            renderComponent({
                onChange,
                selectedId: 1,
            }),
        );

        fireEvent.click(screen.getByTestId(SEGMENT_TEST_ID));

        expect(onChange).toBeCalledWith(2);
    });

    it('should set dataTestId', () => {
        const dataTestId = 'data-test-id';
        const { getByTestId } = render(
            renderComponent({ selectedId: 2, onChange: () => null, dataTestId }),
        );

        expect(getByTestId(dataTestId)).toBeInTheDocument();
    });

    it('should set className', () => {
        const className = 'customClassName';
        const { container } = render(
            renderComponent({
                onChange: () => null,
                className,
                selectedId: 2,
            }),
        );

        expect(container.firstElementChild).toHaveClass(className);
    });

    it('should set custom style', () => {
        const style = { padding: 20 };
        const { container } = render(
            renderComponent({
                onChange: () => null,
                style,
                selectedId: 2,
            }),
        );

        const firstElement = container.firstChild;

        expect(firstElement).toHaveStyle('padding: 20px');
    });
});
