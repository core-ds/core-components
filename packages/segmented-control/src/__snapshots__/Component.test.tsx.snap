// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`segmented-control should display correctly 1`] = `
<div>
  <div>
    <div
      class="wrapper wrapper rectangular size-32"
    >
      <div
        class="container"
      >
        <div
          class="selectedBox selectedBox rectangular"
          style="width: 0px; transform: translateX(0px);"
        />
        <div
          class="inner"
        >
          <button
            class="segment segment segment segment selected"
            type="button"
          >
            Label 1
          </button>
          <button
            class="segment segment segment segment"
            data-test-id="SEGMENT"
            type="button"
          >
            Label 2
          </button>
          <button
            class="segment segment segment segment"
            type="button"
          >
            Label 3
          </button>
          <button
            class="segment segment segment segment"
            type="button"
          >
            Label 4
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
